### PR TITLE
#15 radarr movie lookup

### PR DESCRIPTION
**Pull request by smantic**
adds a function for using radarr's /movie/lookup endpoint

### CODE REVIEW

**Comment by smantic** [comment]
if the golangci-lint config is put in the root of the directory devs can run the correct linters to make that pass ^

File: radarr/radarr.go [hunk 1]
```diff
@@ -431,3 +431,22 @@ func (r *Radarr) SendCommand(cmd *CommandRequest) (*CommandResponse, error) {
...
+// Lookup will search for movies matching the specified search term.
+func (r *Radarr) Lookup(term string) ([]Movie, error) {
+	if term == "" {
+		return nil, nil
+	}
+
+	var out []Movie
+
+	params := make(url.Values)
+	params.Set("term", url.QueryEscape(term))
```
Comment (davidnewhall) [inline radarr/radarr.go:444]: `Encode()` runs `QueryEscape`, see [here](https://cs.opensource.google/go/go/+/refs/tags/go1.17.5:src/net/url/url.go;l=986). That means it'll run twice. The fix is to change this line to:
```suggestion
	params.Set("term", term)
```

File: radarr/radarr.go [hunk 2]
```diff
@@ -431,3 +431,22 @@ func (r *Radarr) SendCommand(cmd *CommandRequest) (*CommandResponse, error) {
...
+
+	err := r.GetInto("v3/movie/lookup", params, &out)
+	if err != nil {
+		return nil, fmt.Errorf("failed to lookup movie: %w", err)
```
Comment (davidnewhall) [inline radarr/radarr.go:448]: Please change this line to match the rest of the library:
```suggestion
		return nil, fmt.Errorf("api.Get(movie/lookup): %w", err)
```

File: radarr/radarr.go [hunk 3]
```diff
@@ -431,3 +431,22 @@ func (r *Radarr) SendCommand(cmd *CommandRequest) (*CommandResponse, error) {
 
 	return &output, nil
 }
+
+// Lookup will search for movies matching the specified search term.
+func (r *Radarr) Lookup(term string) ([]Movie, error) {
```
Comment (davidnewhall) [inline radarr/radarr.go:436]: Does this work OK with the `[]Movie` type? Sometimes there are variations in new methods that must be accounted for. I use this tool a lot to check the consistency of my structs vs the json. https://mholt.github.io/json-to-go/

**Review by davidnewhall — COMMENTED** [review]
Thank you for the contribution! Just a few things to fix, please.

> if the golangci-lint config is put in the root of the directory devs can run the correct linters to make the pipeline pass

You're welcome to do this. It's currently in the [Makefile](https://github.com/golift/starr/blob/41056980c8856485a8598fd886f2cf54e11ead54/Makefile#L12), I run `make test`. As golangci-lint gets updated I have to fix or omit tests.

File: radarr/radarr.go (hunk 3 above)
Comment (smantic) [reply radarr/radarr.go:436]: Here is the generated struct from the response body of this endpoint. It matches `Movie`. 
```go
type AutoGenerated []struct {
	ID              int    `json:"id"`
	Title           string `json:"title"`
	SortTitle       string `json:"sortTitle"`
	SizeOnDisk      int    `json:"sizeOnDisk"`
	Overview        string `json:"overview"`
	InCinemas       string `json:"inCinemas"`
	PhysicalRelease string `json:"physicalRelease"`
	Images          []struct {
		CoverType string `json:"coverType"`
		URL       string `json:"url"`
		RemoteURL string `json:"remoteUrl"`
	} `json:"images"`
	Website             string   `json:"website"`
	Year                int      `json:"year"`
	HasFile             bool     `json:"hasFile"`
	YouTubeTrailerID    string   `json:"youTubeTrailerId"`
	Studio              string   `json:"studio"`
	Path                string   `json:"path"`
	RootFolderPath      string   `json:"rootFolderPath"`
	QualityProfileID    int      `json:"qualityProfileId"`
	Monitored           bool     `json:"monitored"`
	MinimumAvailability string   `json:"minimumAvailability"`
	IsAvailable         bool     `json:"isAvailable"`
	FolderName          string   `json:"folderName"`
	Runtime             int      `json:"runtime"`
	CleanTitle          string   `json:"cleanTitle"`
	ImdbID              string   `json:"imdbId"`
	TmdbID              int      `json:"tmdbId"`
	TitleSlug           string   `json:"titleSlug"`
	Certification       string   `json:"certification"`
	Genres              []string `json:"genres"`
	Tags                []int    `json:"tags"`
	Added               string   `json:"added"`
	Ratings             struct {
		Votes int `json:"votes"`
		Value int `json:"value"`
	} `json:"ratings"`
	Collection struct {
		Name   string `json:"name"`
		TmdbID int    `json:"tmdbId"`
		Images []struct {
			CoverType string `json:"coverType"`
			URL       string `json:"url"`
			RemoteURL string `json:"remoteUrl"`
		} `json:"images"`
	} `json:"collection"`
	Status string `json:"status"`
}
```

Comment (davidnewhall) [reply radarr/radarr.go:436]: awesome. love it when that works out!

**Review by davidnewhall — APPROVED** [review]

**Pull request merged**